src/pngimage.cpp

```diff
@@ -57,6 +57,15 @@ const unsigned char pngBlank[] = { 0x89,0x50,0x4e,0x47,0x0d,0x0a,0x1a,0x0a,0x00,
                                    0x45,0x4e,0x44,0xae,0x42,0x60,0x82
                                  };
 
+namespace
+{
+    inline bool compare(const char* str, const Exiv2::DataBuf& buf, size_t length)
+    {
+        assert(strlen(str) <= length);
```

Comment:
This check would only happen in Debug mode (which is fine for me). Is that what you had in mind?

Reply:
I have added a comment into the code explaining that.